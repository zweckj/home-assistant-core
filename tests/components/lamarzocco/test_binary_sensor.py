"""Tests for La Marzocco binary sensors."""
from unittest.mock import MagicMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

pytestmark = pytest.mark.usefixtures("init_integration")


BINARY_SENSORS = (
    "brew_active",
    "water_reservoir",
)


async def test_binary_sensors(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the La Marzocco binary sensors."""

    serial_number = mock_lamarzocco.serial_number

    for binary_sensor in BINARY_SENSORS:
        state = hass.states.get(f"binary_sensor.{serial_number}_{binary_sensor}")
        assert state
        assert state == snapshot(name=f"{serial_number}_{binary_sensor}-binary_sensor")

        entry = entity_registry.async_get(state.entity_id)
        assert entry
        assert entry.device_id
        assert entry == snapshot(name=f"{serial_number}_{binary_sensor}-entry")
