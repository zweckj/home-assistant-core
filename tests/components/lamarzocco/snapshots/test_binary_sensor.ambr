# serializer version: 1
# name: test_binary_sensors[GS3 AV][GS01234_brew_active-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'GS01234 Brew active',
      'icon': 'mdi:cup-water',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.gs01234_brew_active',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[GS3 AV][GS01234_brew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.gs01234_brew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:cup-water',
    'original_name': 'Brew active',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'brew_active',
    'unique_id': 'GS01234_brew_active',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[GS3 AV][GS01234_water_reservoir-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'problem',
      'friendly_name': 'GS01234 Water reservoir',
      'icon': 'mdi:water-well',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.gs01234_water_reservoir',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[GS3 AV][GS01234_water_reservoir-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.gs01234_water_reservoir',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.PROBLEM: 'problem'>,
    'original_icon': 'mdi:water-well',
    'original_name': 'Water reservoir',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_reservoir',
    'unique_id': 'GS01234_water_reservoir',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[GS3 MP][GS01234_brew_active-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'GS01234 Brew active',
      'icon': 'mdi:cup-water',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.gs01234_brew_active',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[GS3 MP][GS01234_brew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.gs01234_brew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:cup-water',
    'original_name': 'Brew active',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'brew_active',
    'unique_id': 'GS01234_brew_active',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[GS3 MP][GS01234_water_reservoir-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'problem',
      'friendly_name': 'GS01234 Water reservoir',
      'icon': 'mdi:water-well',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.gs01234_water_reservoir',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[GS3 MP][GS01234_water_reservoir-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.gs01234_water_reservoir',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.PROBLEM: 'problem'>,
    'original_icon': 'mdi:water-well',
    'original_name': 'Water reservoir',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_reservoir',
    'unique_id': 'GS01234_water_reservoir',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[Linea Mini][LM01234_brew_active-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'LM01234 Brew active',
      'icon': 'mdi:cup-water',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.lm01234_brew_active',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[Linea Mini][LM01234_brew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.lm01234_brew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:cup-water',
    'original_name': 'Brew active',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'brew_active',
    'unique_id': 'LM01234_brew_active',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[Linea Mini][LM01234_water_reservoir-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'problem',
      'friendly_name': 'LM01234 Water reservoir',
      'icon': 'mdi:water-well',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.lm01234_water_reservoir',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[Linea Mini][LM01234_water_reservoir-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.lm01234_water_reservoir',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.PROBLEM: 'problem'>,
    'original_icon': 'mdi:water-well',
    'original_name': 'Water reservoir',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_reservoir',
    'unique_id': 'LM01234_water_reservoir',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[Micra][MR01234_brew_active-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'MR01234 Brew active',
      'icon': 'mdi:cup-water',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.mr01234_brew_active',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[Micra][MR01234_brew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.mr01234_brew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:cup-water',
    'original_name': 'Brew active',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'brew_active',
    'unique_id': 'MR01234_brew_active',
    'unit_of_measurement': None,
  })
# ---
# name: test_binary_sensors[Micra][MR01234_water_reservoir-binary_sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'problem',
      'friendly_name': 'MR01234 Water reservoir',
      'icon': 'mdi:water-well',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.mr01234_water_reservoir',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_binary_sensors[Micra][MR01234_water_reservoir-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.mr01234_water_reservoir',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.PROBLEM: 'problem'>,
    'original_icon': 'mdi:water-well',
    'original_name': 'Water reservoir',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_reservoir',
    'unique_id': 'MR01234_water_reservoir',
    'unit_of_measurement': None,
  })
# ---
